source/jmap/types.d

```diff
@@ -877,217 +873,57 @@ struct Invocation {
+    alias filter this;
 }
 
-
-package enum nullArray = (Nullable!(string[])).init;
-package enum NullUint = (Nullable!uint).init;
-package enum NullDateTime = (Nullable!DateTime).init;
-
-FilterCondition filterCondition(string inMailbox = null,
-        Nullable!(string[])inMailboxOtherThan = nullArray,
-        string before = null,
-        string after = null,
-        Nullable!uint minSize = NullUint,
-        Nullable!uint maxSize = NullUint,
-        string allInThreadHaveKeyword = null,
-        string someInThreadHaveKeyword = null,
-        string noneInThreadHaveKeyword = null,
-        string hasKeyword = null,
-        string notKeyword = null,
-        string text = null,
-        string from = null,
-        string to = null,
-        string cc = null,
-        string bcc = null,
-        string subject = null,
-        string body_ = null,
-        Nullable!(string[])header = nullArray, ) {
-    return new FilterCondition(inMailbox, inMailboxOtherThan, before, after, minSize,
-            maxSize, allInThreadHaveKeyword, someInThreadHaveKeyword, noneInThreadHaveKeyword,
-            hasKeyword, notKeyword, text, from, to, cc, bcc, subject, body_, header);
-}
-
-private void doSerialize(S, T)(ref S serializer, T t) {
-    import std.traits : hasUDA, getUDAs, isCallable, isInstanceOf;
-    auto o = serializer.objectBegin;
-    static foreach (i, M; T.tupleof) {
-        {
-            static if (!isCallable!M) {
-                static if (hasUDA!(M, "serdeKeys")) {
-                    enum name = getUDAs!(M, "serdeKeys")[0].value;
-                } else {
-                    enum name = __traits(identifier, M);
-                }
-                mixin("auto value = t." ~ __traits(identifier, M) ~ ";");
-                static if (isInstanceOf!(Nullable, typeof(value))) {
-                    if (!value.isNull) {
-                        // static if (is(typeof(value.get.result) == typeof(Variable)))
-                        //      serializeAsAsdf(value.get.result);
-                        // else static if is(Unqual!
-                        //      doSerialize(serializer,value.get.result);
-                        serializer.serializeAsdf(serializeToAsdf(value.get));
-                    } else {
-                        serializer.putValue(null);
-                    }
-                } else {
-                    serializer.serializeAsdf(serializeToAsdf(result));
-                }
-            }
-        }
-    }
-}
-
-string toUTCDate(Nullable!DateTime dt) {
-    import std.exception : enforce;
-    enforce(!dt.isNull, "datetime must not be null");
-    return toUTCDate(dt.get);
-}
-
-// "2014-10-30T06:12:00Z"
-string toUTCDate(DateTime dt) {
-    import std.string : format;
-    return format!"%04d-%02d-%02dT%02d:%02d:%02dZ"(
-        dt.date.year,
-        dt.date.month,
-        dt.date.day,
-        dt.timeOfDay.hour,
-        dt.timeOfDay.minute,
-        dt.timeOfDay.second,
-    );
+struct FilterOperators {
```

Comment:
Shouldn't this be called `FilterOperator` ?

Reply:
Thanks, that is typo